src/MySqlConnector/Protocol/Serialization/SlicedSurrogatesUtility.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MySqlConnector.Protocol.Serialization
+{
+#if !NET45 && !NET461 && !NET471 && !NETSTANDARD1_3 && !NETSTANDARD2_0 && !NETSTANDARD2_1 && !NETCOREAPP2_1
+
+	internal static class SlicedSurrogatesUtility
+	{
+		public static int GetUtf8ByteCount(ReadOnlySpan<char> span, ref char? surrogatePartBuffer)
+		{
+			var containsSurrogateAtBeginning = BitConverter.IsLittleEndian
+				? char.IsLowSurrogate(span[0])
+				: char.IsHighSurrogate(span[0]);
+			var containsSurrogateAtEnding = BitConverter.IsLittleEndian
+				? char.IsHighSurrogate(span[^1])
+				: char.IsLowSurrogate(span[^1]);
```

Comment:
```suggestion
			var containsSurrogateAtBeginning = char.IsLowSurrogate(span[0]);
			var containsSurrogateAtEnding = char.IsHighSurrogate(span[^1]);
```

`IsLittleEndian` shouldn't make any difference here: the high-surrogate code unit is always stored first, followed by the low-surrogate code unit.

(If these were serialised to bytes, the byte order of the UTF-16 code units themselves could be in little-endian or big-endian byte order, but that's irrelevant when dealing with `Span<char>`.)

This should just be collapsed down to the case where `IsLittleEndian == true`.